### PR TITLE
UIEH-916: reset router state when visiting eholdings

### PR DESCRIPTION
**Pull request by MaksymDryha**
We have routing issues caused by preserving the router state when navigating outside of the eholdings app.  This PR fixes these issues by resetting the router state after the eholdings root component is mounted (if eholdings app is being mounted, than it is safe to say that previously we were in another application and thus, the eholdings app router state should be reset).

[Ticket](https://issues.folio.org/browse/UIEH-916)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -31,6 +31,13 @@ import SettingsAssignedUsersRoute from './routes/settings-assigned-users-route';
 
 class EHoldings extends Component {
   static propTypes = {
+    history: PropTypes.shape({
```
Comment (BogdanDenis) [inline src/index.js:34]: I think we should use `react-router-prop-types` here to handle the shape of `history`, `location` and `match` props

File: src/components/details-view/details-view.js [hunk 2]
```diff
@@ -208,8 +208,15 @@ class DetailsView extends Component {
       location,
     } = this.props;
 
+    const previousEHoldingsLocationKnown = location.search.includes('searchType');
+
     if (location.state && location.state.eholdings) {
       history.goBack();
+    } else if (previousEHoldingsLocationKnown) {
```
Comment (BogdanDenis) [inline src/components/details-view/details-view.js:215]: Please add a test to cover this case

**Comment by MaksymDryha** [comment]
> Please, update changelog

@Vladyslav-Velytskyi , thanks, I'm always forgetting about changelog🤦

File: src/components/details-view/details-view.js (hunk 2 above)
Comment (MaksymDryha) [reply src/components/details-view/details-view.js:215]: @BogdanDenis , done

**Pull request merged**